### PR TITLE
Use tokenizer for parsing as well

### PR DESCRIPTION
**Pull request by Aloso**
This is a rather big change, aiming to simplify parsing. Here's a summary:

* `tokenizer.rs` was refactored to use a struct containing a `TokenKind` instead of an enum. This makes it easier to match on the token's string content

* the tokenizer now also parses comments. This improves syntax highlighting.

* the tokenizer now also parses invalid characters. These are highlighted red.

* the `parse_script` function now tokenizes the input and removes all whitespace and comment tokens, and then proceeds with parsing. This improves the parser for several reasons:
  * no need to account for whitespace in the parser
  * no need to check if keywords or number literals are valid in the parser
  * robustness, separation of concerns
  * consistency: the syntax highlighter is guaranteed to have the same behaviour as the parser
  * no code duplication

* all parser functions in `parser.rs` now accept and return a new `Tokens<'a>` type, which is a wrapper for `&'a [Token<'a>]`.

* usages of the `tag` parser have been replaced with a new `text` parser. That parser checks if the text in the next token matches the provided string. It doesn't check if the `TokenKind` is correct, but that is usually not necessary (for example, a token with the text `"for"` is guaranteed to be a keyword). Note that this was quite error-prone before, because `tag("for")` also parses the text `"forager"`

* to make sure that no invalid numbers are parsed (such as `1.05px`, which is tokenized as `FloatLiteral("1.05")`, `Symbol("px")`), the `parse_script` function checks that "symbol-like" tokens can't appear directly after another without whitespace or punctuation in between.

  "symbol-like" tokens are number literals, bool literals, keywords, operators and symbols, because they can all contain the same characters.

* because the tokenizer parses comments, the `comment` crate was removed from dependencies

P.S. This also slightly changes how `=` is parsed: It is now nowhere required to put spaces around the `=` sign, so `let foo={ bar=baz }` works fine.

### CODE REVIEW

**Comment by adam-mcdaniel** [comment]
When using the `==` operator to check for equality this seems to break :(
image

File: src/tokenizer.rs [hunk 1]
```diff
@@ -53,7 +45,8 @@ fn any_punctuation(input: &str) -> IResult<&str, &'static str, ()> {
         punctuation_tag("@"),
         punctuation_tag("\'"),
         punctuation_tag(","),
-        keyword_tag("="),  // must be surrounded by whitespace
+        punctuation_tag(";"),
+        punctuation_tag("="),
```
Comment (adam-mcdaniel) [inline src/tokenizer.rs:51]: Maybe add a `punctuation_tag("==")` here prior to `punctuation_tag("=")`?

**Comment by Aloso** [comment]
It's fixed now.

**Comment by Aloso** [comment]
Don't merge yet, I might have found another bug.

**Comment by Aloso** [comment]
Nevermind, this is good to go!

**Comment by adam-mcdaniel** [comment]
Absolutely wonderful! I love the syntax highlighting for comments!!! This is amazing work, thank you :) ❤️

**Pull request merged**